### PR TITLE
Add test coverage for PropertyGrid

### PR DESCRIPTION
**Pull request by MelonWang1**
Related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for PropertyGrid to test its events: SelectedGridItemChanged, PropertyTabChanged.
- Add unit tests for PropertyGridToolStripAccessibleObject to test its methods: Name.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11587)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/PropertyGridToolStrip.PropertyGridToolStripAccessibleObjectTests.cs [hunk 1]
```diff
@@ -47,4 +47,22 @@ public void PropertyGridToolStripAccessibleObject_Role_IsToolBar_ByDefault()
...
     }
+
+    [WinFormsTheory]
+    [InlineData("Test Name", null, "Test Name")]
+    [InlineData(null, "Parent Name", "Parent Name")]
+    public void PropertyGridToolStripAccessibleObject_Name_ReturnsExpected(string toolStripAccessibleName, string parentAccessibleName, string expectedName)
+    {
+        using PropertyGrid propertyGrid = new();
+        propertyGrid.AccessibleName = parentAccessibleName;
+        using var propertyGridToolStrip = new PropertyGridToolStrip(propertyGrid);
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/PropertyGridToolStrip.PropertyGridToolStripAccessibleObjectTests.cs:58]: Using new type:
using PropertyGridToolStrip propertyGridToolStrip = new(propertyGrid);

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/PropertyGridToolStrip.PropertyGridToolStripAccessibleObjectTests.cs [hunk 2]
```diff
@@ -47,4 +47,22 @@ public void PropertyGridToolStripAccessibleObject_Role_IsToolBar_ByDefault()
...
+        propertyGridToolStrip.AccessibleName = toolStripAccessibleName;
+        var accessibleObject = new PropertyGridToolStrip.PropertyGridToolStripAccessibleObject(propertyGridToolStrip, propertyGrid);
+
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/PropertyGridToolStrip.PropertyGridToolStripAccessibleObjectTests.cs:61]: using new type:
PropertyGridToolStrip.PropertyGridToolStripAccessibleObject accessibleObject = new(propertyGridToolStrip, propertyGrid);

**Review by ricardobossan — COMMENTED** [review]
LGTM when addressed colleagues comments

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs [hunk 3]
```diff
@@ -4068,6 +4068,62 @@ public class TestObject
...
+        propertyGrid.PropertyTabChanged += (sender, e) =>
+        {
+            eventCallCount++;
+            eventArgs = e;
+        };
+
+        var tab1 = new TestPropertyTab();
+        var tab2 = new TestPropertyTab();
+        propertyGrid.PropertyTabs.AddTabType(tab2.GetType());
+        propertyGrid.PropertyTabs.AddTabType(tab1.GetType());
```
Comment (Epica3055) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs:4087]: I tried on my machine and found that these two lines are not necessary. 🤔

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs [hunk 4]
```diff
@@ -4068,6 +4068,60 @@ public class TestObject
...
+        PropertyTabChangedEventArgs eventArgs = null;
+
+        propertyGrid.PropertyTabChanged += (sender, e) =>
+        {
+            eventCallCount++;
+            eventArgs = e;
+        };
+
+        var tab1 = new TestPropertyTab();
+        var tab2 = new TestPropertyTab();
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs:4085]: Please use target type new
```suggestion
        TestPropertyTab tab1 = new();
        TestPropertyTab tab2 = new();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs [hunk 5]
```diff
@@ -4068,6 +4068,60 @@ public class TestObject
...
+        object actualSender = null;
+
+        propertyGrid.SelectedGridItemChanged += (sender, e) =>
+        {
+            eventCallCount++;
+            eventArgs = e;
+            actualSender = sender;
+        };
+
+        var gridItemMock = new Mock<GridItem>();
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs:4113]:
```suggestion
        Mock<GridItem>gridItemMock = new();
```

**Review by lonitra — APPROVED** [review]

**Pull request merged**